tools/doc/html.js

```diff
@@ -205,10 +205,19 @@ function preprocessElements(lexed) {
           headingIndex = -1;
           heading = null;
         }
-        token.text = `<div class="api_stability api_stability_${number}">` +
-          '<a href="documentation.html#documentation_stability_index">' +
-          `${prefix} ${number}</a>${explication}</div>`
-          .replace(/\n/g, ' ');
+
+        // Do not link to the section we are already in.
+        if (filename === 'documentation' &&
+            heading && heading.text === 'Stability Index') {
+          token.text = `<div class="api_stability api_stability_${number}">` +
+            `${prefix} ${number}${explication}</div>`
+            .replace(/\n/g, ' ');
+        } else {
+          token.text = `<div class="api_stability api_stability_${number}">` +
+            '<a href="documentation.html#documentation_stability_index">' +
+            `${prefix} ${number}</a>${explication}</div>`
+            .replace(/\n/g, ' ');
+        }
```

Comment:
I suggest to change this to something like:

```js
const link = filename === 'documentation' &&
  heading && heading.text === 'Stability Index' ?
    `${prefix} ${number}` :
    '<a href="documentation.html#documentation_stability_index">' +
      `${prefix} ${number}</a>`;
token.text = `<div class="api_stability api_stability_${number}">` +
  `${link}${explication}</div>`
    .replace(/\n/g, ' ');
```

It is just a bit shorter due to having less repetition.